### PR TITLE
SC coverage (3/3)

### PR DESCRIPTION
**Pull request by matthew7838**
### Summary
This is the last part of coverage to ensure 100% test coverage on the Soil and Crop branch. There will still be another PR to continuously hunts down some uncovered code throughout the debugging process and changes to the tests.

### Changes
- Added tests cases and conditions to test when layering temperature <=0 in mineralize_organic_nitrogen() for humus_mineralization module.
- Created a file to simply test `NitrogenCycle`'s main routine
- `percolate_between_layers` in `percolation` are now tested completely differently
- added test cases to fully cover the `percolation` main routine
- added a new file to test the phosphorus cycle main routine
- Created a file to test all methods of `soil.py`
- `soil.py` now removes the config method since it's not used and probably won't be
- `soil_data.py` is now added with test cases when errors should be thrown, removed statements to check if layer data is none, and `cover_factor` is fully tested.
- Added tests for `test_determine_support_practice_factor` and added cases when an error should be thrown in the main routine of `soil_erosion.py`

### CODE REVIEW

File: SC_redesign/Crop_and_Soil/soil/percolation.py [hunk 1]
```diff
@@ -36,6 +36,7 @@ def percolate(self, has_seasonal_high_water_table: bool) -> None:
         deepest_layer = layer_count - 1
 
         for layer_number in range(layer_count):  # loop through each layer
+            print(layer_number)
```
Comment (Isgotkowitz) [inline SC_redesign/Crop_and_Soil/soil/percolation.py]:
```suggestion
```

File: SC_redesign/Crop_and_Soil/soil/phosphorus_cycling/soluble_phosphorus.py [hunk 2]
```diff
@@ -188,8 +188,11 @@ def _determine_dissolved_reactive_phosphorus_leachate(soil_phosphorus: float, is
         """
         try:
             dissolved_reactive_phosphorus_leachate = exp((soil_phosphorus * 1.5 - isotherm_intercept) / isotherm_slope)
+            print("did not fail")
         except OverflowError:
+            print("a")
             dissolved_reactive_phosphorus_leachate = inf
+        print(dissolved_reactive_phosphorus_leachate)
```
Comment (Isgotkowitz) [inline SC_redesign/Crop_and_Soil/soil/phosphorus_cycling/soluble_phosphorus.py]:
```suggestion
        except OverflowError:
            dissolved_reactive_phosphorus_leachate = inf
```

File: SC_redesign/Crop_and_Soil/tests_SC/soil_tests/phosphorus_cycling_tests/test_soluble_phosphorus.py [hunk 3]
```diff
@@ -59,16 +60,22 @@ def test_determine_isotherm_intercept(slope: float) -> None:
...
+                                                          overflow: bool) -> float:
     """Tests that the amount of phosphorus calculated to leach out of the layer is correct."""
-    observed = SolublePhosphorus._determine_dissolved_reactive_phosphorus_leachate(phosphorus, slope, intercept)
-    expected = min(20.0, exp(((phosphorus * 1.5) - intercept) / slope))
-    assert observed == expected
+    if overflow:
+        observed = SolublePhosphorus._determine_dissolved_reactive_phosphorus_leachate(phosphorus, slope, intercept)
+        assert observed == 20
+    else:
+        observed = SolublePhosphorus._determine_dissolved_reactive_phosphorus_leachate(phosphorus, slope, intercept)
```
Comment (Isgotkowitz) [inline SC_redesign/Crop_and_Soil/tests_SC/soil_tests/phosphorus_cycling_tests/test_soluble_phosphorus.py]:
```suggestion
    observed = SolublePhosphorus._determine_dissolved_reactive_phosphorus_leachate(phosphorus, slope, intercept)
    if overflow:
        assert observed == 20
    else:
```

**Review by Isgotkowitz — APPROVED** [review]
Looks great Matthew. Only a few small suggestions

File: SC_redesign/Crop_and_Soil/soil/phosphorus_cycling/soluble_phosphorus.py (hunk 2 above)
Comment (matthew7838) [reply SC_redesign/Crop_and_Soil/soil/phosphorus_cycling/soluble_phosphorus.py]: another good catch, thanks again

File: SC_redesign/Crop_and_Soil/tests_SC/soil_tests/phosphorus_cycling_tests/test_soluble_phosphorus.py (hunk 3 above)
Comment (matthew7838) [reply SC_redesign/Crop_and_Soil/tests_SC/soil_tests/phosphorus_cycling_tests/test_soluble_phosphorus.py]: a lot cleaner, thanks.

**Review by ew3361zh — APPROVED** [review]
LGTM!

**Pull request merged**